mitodl/ocw-hugo-themes
# Update Prettier for better TS support.

#### Pre-Flight checklist


#### What are the relevant tickets?
#973 

#### What's this PR do?
This PR is a major version change of Prettier (v2)
- v2 supports more typescript syntax (notably, `import type` and `export type`, which v1 does not support)
- There are some formatting changes from v1 to v2 of prettier. I sort of prefer some of the v1 formatting, but mostly I don't care and I want support for importing types. If anyone does care, we can presumably use some eslint rule to override prettier's choices. But not worth the effort imo.

#### How should this be manually tested?
1. Introduce a formatting error somewhere and save (depending on editor + extensions, may need to "Save without formatting").
2. Run `yarn fmt:check`
3. Run `echo $?` to view the exit code of the previous process. It should be 1.
